### PR TITLE
Handle 406 errors apart from generic Carina errors

### PR DESCRIPTION
**Pull request by thomasem**

### CODE REVIEW

**Comment by carolynvs** [comment]
Looks good, have a 👍 after you get a ✅

**Comment by thomasem** [comment]
@carolynvs Ready for final review.

File: libcarina_test.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	if httpErr, ok := cause.(HTTPErr); ok {
+		if httpErr.StatusCode != 406 {
+			t.Error("expected StatusCode of 406, got", httpErr.StatusCode)
+		}
+	} else {
+		t.Error("expected to get HTTPErr, got", reflect.TypeOf(err))
+	}
+}
+
+func TestMicroversionUnsupportedNewRequest(t *testing.T) {
```
Comment (carolynvs) [inline libcarina_test.go:67]: Is this still needed now that we have tests on get/list/etc?

File: libcarina_test.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		t.FailNow()
+	}
+	resp, err := carinaClient.GetCredentials("9f18f7f9-aeb4-4c7c-91ef-e13ff94e352c")
+	if resp != nil {
+		t.Error("expected nil response, got", resp)
+	}
+	assertMicroversionUnsupportedHandled(t, err)
+}
+
+func TestMicroversionUnsupportedGetAPIMetadata(t *testing.T) {
```
Comment (carolynvs) [inline libcarina_test.go]: I would lean towards removing this as part of the story, since it's not used anymore.

**Review by carolynvs — CHANGES_REQUESTED** [review]

File: libcarina_test.go (hunk 2 above)
Comment (thomasem) [reply libcarina_test.go]: It still represents bindings for a particular portion of the API. Are you suggesting I delete the function from the library?

File: libcarina_test.go (hunk 1 above)
Comment (thomasem) [reply libcarina_test.go:67]: I think it's some extra assurance about all things calling the Carina API. Sort of asserts that the expectations of `NewRequest` for other functions that're using it. That way you can tell, if you start seeing failures, whether it's in `NewRequest`, or something calling it.

Comment (carolynvs) [reply libcarina_test.go:67]: A+, makes sense to me

File: libcarina_test.go (hunk 2 above)
Comment (carolynvs) [reply libcarina_test.go]: Yeah, it was always the plan to nuke that when we had the 406 responses out of the API.

Comment (thomasem) [reply libcarina_test.go]: Fixed

**Review by carolynvs — APPROVED** [review]

**Comment by carolynvs** [comment]
:shipit:

**Pull request merged**